libs/category/src/models/generic-category.ts

```diff
@@ -11,12 +11,36 @@ import { DaffCategoryBreadcrumb } from './category-breadcrumb';
  * this type is a recursive generic.
  */
 export interface DaffGenericCategory<T extends DaffGenericCategory<T>> extends DaffLocatable {
+	/**
+	 * The id of the category.
+	 */
   id: ID;
+	/**
+	 * The name of the category that can be used as a UI label.
+	 */
 	name: string;
+	/**
+	 * A description of the category.
+	 */
 	description?: string;
+	/**
+	 * The number of child category nodes under this category.
+	 */
   children_count?: number;
+	/**
+	 * The number of products described by this category.
+	 */
   total_products?: number;
+	/**
+	 * The direct child categories of this category.
```

Comment:
`the category`?

Reply:
This is the same situation as the other discussion we were having: 
> I'd rather use "this" when there are multiple categories "the" could be referring to. I could probably use "the" and presume that the reader will correctly assume I mean "this" (and they probably would), but I could also just use "this".